.github/workflows/update-dev-image.yaml

```diff
@@ -0,0 +1,28 @@
+name: Build Dev Container Image & Push To GitHub Container Registry
+
+on:
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build-and-push-container-image:
```

Comment:
The job name should be unique from the normal build action. Maybe `build-and-push-dev-container-image`?